File: api/probe-mara.js
module.exports = async function handler(req, res) {
  try {
    const r = await fetch("https://www.mara.gov.om/calendar_page2.asp", {
      headers: { "User-Agent": "Mozilla/5.0 (compatible; OmanPrayerTimesMirror/1.0)" },
    });
    const html = await r.text();

    const forms = [...html.matchAll(/<form\b[\s\S]*?<\/form>/gi)].map(m => m[0]);
    const compact = forms.map((form, i) => ({
      index: i,
      openTag: (form.match(/<form\b[^>]*>/i) || [""])[0],
      selects: [...form.matchAll(/<select\b[^>]*>[\s\S]*?<\/select>/gi)].map(x => x[0]),
      inputs: [...form.matchAll(/<input\b[^>]*>/gi)].map(x => x[0]),
    }));

    res.status(200).json({
      upstreamStatus: r.status,
      finalUrl: r.url,
      formCount: forms.length,
      forms: compact,
      head: html.slice(0, 4000),
    });
  } catch (e) {
    res.status(500).json({ error: String(e && e.stack || e) });
  }
};

File: scripts/probe_mara.py
from pathlib import Path
import requests

URL = "https://www.mara.gov.om/calendar_page2.asp"
headers = {"User-Agent": "Mozilla/5.0 (compatible; OmanPrayerTimesMirror/1.0)"}

r = requests.get(URL, headers=headers, timeout=30)
r.raise_for_status()
Path("mara_probe.html").write_text(r.text, encoding="utf-8")
print("status", r.status_code)
print("url", r.url)
print("bytes", len(r.text.encode("utf-8")))
print(r.text[:20000])
